spark-operator-api/src/main/java/org/apache/spark/k8s/operator/spec/RestartPolicy.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.spec;
+
+import org.apache.spark.k8s.operator.status.BaseStateSummary;
+
+public enum RestartPolicy {
+  Always,
+  Never,
+  OnFailure,
+  OnInfrastructureFailure;
+
+  public static boolean attemptRestartOnState(
+      final RestartPolicy policy, final BaseStateSummary stateSummary) {
```

Comment:
Given the usage of this method, please use `JEP 361: Switch Expressions` to simplify the body, @jiangzho .
- https://openjdk.org/jeps/361

```java
-    switch (policy) {
-      case Always:
-        return true;
-      case Never:
-        return false;
-      case OnFailure:
-        return stateSummary.isFailure();
-      case OnInfrastructureFailure:
-        return stateSummary.isInfrastructureFailure();
-    }
-    return false;
+    return switch (policy) {
+      case Always -> true;
+      case Never -> false;
+      case OnFailure -> stateSummary.isFailure();
+      case OnInfrastructureFailure -> stateSummary.isInfrastructureFailure();
+      default -> false;
+    };
```